ci-operator/step-registry/aws/deprovision/iam-profile/aws-deprovision-iam-profile-commands.sh

```diff
@@ -10,115 +10,105 @@ trap 'delete_all' EXIT TERM INT
-		if [ X"$policy_arn" == X"" ]; then
-			continue
-		fi
-		echo -e "detaching policy: ${policy_arn}"
-		aws --region $aws_region iam detach-role-policy --role-name ${role_name} --policy-arn ${policy_arn} || return 1
-	done
-
-	# delete inline policy
-	inline_policies=$(aws --region $aws_region iam list-role-policies --role-name ${role_name} | jq -r .PolicyNames[])
-	for policy_name in $inline_policies; do
-		if [ X"$policy_name" == X"" ]; then
-			continue
-		fi
-		echo -e "deleting inline policy: ${policy_name}"
-		aws --region $aws_region iam delete-role-policy --role-name ${role_name} --policy-name ${policy_name} || return 1
-	done
-
-	aws --region $aws_region iam delete-role --role-name ${role_name} || return 1
-	echo -e "deleted role: ${role_name}"
-
-	return 0
+  local aws_region=$1
+  local role_name=$2
+
+  echo -e "Deleteing role: $role_name"
+  # detach policy
+  attached_policies=$(aws --region $aws_region iam list-attached-role-policies --role-name ${role_name} | jq -r .AttachedPolicies[].PolicyArn)
+  for policy_arn in $attached_policies; do
+    if [ X"$policy_arn" == X"" ]; then
+      continue
+    fi
+    echo -e "\tDetaching policy: ${policy_arn}"
+    aws --region $aws_region iam detach-role-policy --role-name ${role_name} --policy-arn ${policy_arn} || return 1
+  done
+
+  # delete inline policy
+  inline_policies=$(aws --region $aws_region iam list-role-policies --role-name ${role_name} | jq -r .PolicyNames[])
+  for policy_name in $inline_policies; do
+    if [ X"$policy_name" == X"" ]; then
+      continue
+    fi
+    echo -e "\tDeleting inline policy: ${policy_name}"
+    aws --region $aws_region iam delete-role-policy --role-name ${role_name} --policy-name ${policy_name} || return 1
+  done
+
+  aws --region $aws_region iam delete-role --role-name ${role_name} || return 1
+  echo -e "\tDeleted."
+
+  return 0
 }
 
 function aws_delete_profile() {
-	local aws_region=$1
-	local profile_name=$2
-
-	echo -e "deleting profile: $profile_name"
-	# detach role
-	echo -e "getting attached roles..."
-	attached_roles=$(aws --region $aws_region iam get-instance-profile --instance-profile-name ${profile_name} | jq -r .InstanceProfile.Roles[].RoleName)
-	for role_name in $attached_roles; do
-		if [ X"$role_name" == X"" ]; then
-			continue
-		fi
-		echo -e "detaching role: ${role_name}"
-		aws --region $aws_region iam remove-role-from-instance-profile --instance-profile-name ${profile_name} --role-name ${role_name}
-	done
-
-	aws --region $aws_region iam delete-instance-profile --instance-profile-name ${profile_name}
-	echo -e "deleted profile: ${profile_name}"
-
-	return 0
+  local aws_region=$1
+  local profile_name=$2
+
+  echo -e "Deleting profile: $profile_name"
+  # detach role
+  attached_roles=$(aws --region $aws_region iam get-instance-profile --instance-profile-name ${profile_name} | jq -r .InstanceProfile.Roles[].RoleName)
+  for role_name in $attached_roles; do
+    if [ X"$role_name" == X"" ]; then
```

Comment:
In the `for` loop, "role_name" would never be empty, right, sounds like we do not need this if condition.

Reply:
Yes, but this piece of code is very old, I don't remember if I met the output format exceptions that caused `role_name` is empty.

Reply:
@yunjiang29 do you mind we have a try to remove the `if` condition to see what happen ?

Reply:
OK